server.js

```diff
@@ -0,0 +1,93 @@
+
+require('dotenv').config();
+
+const express = require('express');
+const cors = require('cors');
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcryptjs');
+const db = require('./data/dbConfig.js');
+
+const app = express();
+
+app.use(express.json());
+app.use(cors());
+
+function generateToken(user) {
+  const payload = {
+    subject: user.id,
+    username: user.username,
+    department: user.department,
+  }
+  const secret = process.env.JWT_SECRET;
+  const options = {
+    expiresIn: '1hr'
+  }
+  return jwt.sign(payload, secret, options)
+}
+
+app.post('/api/register', (req, res) => {
+  const creds = req.body;
+
+  const hash = bcrypt.hashSync(creds.password, 10);
+
+  creds.password = hash;
+
+  db('users').insert(creds).then(ids => {
+    res.status(201).json(ids);
+  }).catch(err => json(err));
```

Comment:
You are not handling the error properly, it will throw `json undefined` error and scream about unhandled rejections. I believe this is not a promise on it's own.